### PR TITLE
Section 30 rev2

### PR DESCRIPTION
**Pull request by canyon289**
Added an expanded outline for section 30

I also added a brief summary of *what I think* section 40 and the lesson recap would be so I could "estimate the trajectory" of the narrative arc in context, but I am not tied to them. Similar to a GP want to consider the weight just not the most immediate "data point" but the next two as well ;)

Let me know what you think. Ill be sure not to expand Section 30 out into a full section until you approve

### CODE REVIEW

File: lesson_2_Section_30_new.ipynb [hunk 1]
```diff
@@ -0,0 +1,237 @@
+{
```
Comment (bwengals) [inline lesson_2_Section_30_new.ipynb:1]: <p>I think you dont necessarily have to be 5 for the analogy to work right?  I think if you're an impressionist painter drawing the line you'd probably get the same result as a 5 year old.  There could be a small chance it comes across a bit patronizing.  Overall though I like your point.  Maybe like, "Wether you're 5 years old or an *insert fancy type of artist here*, you don't think about drawing like, here's a fixed function, now let's mentally minimize the residual"? though that's maybe too wordy.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (bwengals) [inline lesson_2_Section_30_new.ipynb:1]: <p>I'm envisioning 4 proposed dots here for Jan 1 2022 right?  I think this is perfect</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (bwengals) [inline lesson_2_Section_30_new.ipynb:1]: <p>If you extend to all X you'd get a flat line that's just the average of everything, right?  Or are you doing like a sliding window?  I think it makes it super clear that, one data point on either side isnt enough data points, all X is too many data points (flat line), how do you pick the right number?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (bwengals) [inline lesson_2_Section_30_new.ipynb:1]: <p>Good place to end the section</p><p> </p><p>Small thing, I realized Im not 100% sure what exactly is the windowed averaging that'll be shown. Do we wan't to call it that or call it nearest neighbors?  Might be nice to call it that, since it's googleable.  Could make an exercise to implement k nearest neighbor regression from scratch in python since its easy, or use scikit-learn on some data, or both and compare?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (bwengals) [inline lesson_2_Section_30_new.ipynb:1]: <p>Yes this outline is exactly right.  Planning on following this exactly.  The last point </p><p> </p><blockquote>What a full GP would look like is this but as you can see there's a lot more going on. Thatll be for the next lesson</blockquote><p> </p><p>is a great place to end</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='16'/>

Comment (bwengals) [inline lesson_2_Section_30_new.ipynb:1]: <p>Want to verify, is the end of Sec 40 the end of this course, or like, the demo/trial/teaser or for free version of the course/lesson?  I could imagine it working that way, since you've get a little intuitive value built up with the kernels and nearest neighbor comparison and then there's a bit of a full GP cliffhanger. Maybe this far is enough that people feel like they learned something and then would be willing to shell out some $ for the rest.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bwengals/etc/pull/5/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

**Comment by bwengals** [comment]
Hey I think it really looks awesome.  Comments I left are pretty much clarifications/verification sort of questions, so please feel free to keep going w it

File: lesson_2_Section_30_new.ipynb (hunk 1 above)
Comment (canyon289) [reply lesson_2_Section_30_new.ipynb:1]: Oh youre right I was loose with my terminology but it does open an interesting question, should we show nearest neighbors or a window?

Window might be more intuitive for introducing kernel because you just extend the window to infinity and add a weighting function, but then I guess you could say the same for N neighbors as well

And technically were not saying nearest neighbors but N to the left and right

Anyhow between the three whats your strongest preference
1. Window function
2. Nearest Neighbors
3. N closest left and right?

Mine is on 1, but only weakly so. I think 3 we should avoid entirely

Comment (canyon289) [reply lesson_2_Section_30_new.ipynb:1]: 4 is what I had in mind yep

Comment (canyon289) [reply lesson_2_Section_30_new.ipynb:1]: good point, ill change this to impressionist painter, itl make for a better thing add as a fun image anyway

Comment (canyon289) [reply lesson_2_Section_30_new.ipynb:1]: Section 40 is definitely not the end of the course, we need to actually explain what GPs are.

I should have been more clear. The primary objective of all these courses is to give students the most practical and intuitive way to learn a concept. With that objective in mind I think this lesson does a great job easing people into the course, starting people with a familiar concept, showing them a bayesian fit as a refresher, and then explaining why functional models, even with bayesian parameter estimates, have challenges, and theres this other way of thinking about observed data fitting. 

That then motivates people to go through all the hard stuff of MVN and Covariance that we need to explain right after this. 
As a secondary benefit I think this just so happens to also work out as a good stopping point for the free lesson, though we can make that decision later.

Does that framing make sense?

Comment (bwengals) [reply lesson_2_Section_30_new.ipynb:1]: I see, that makes sense, def agree.  It's the stopping point of the free *lesson* not course.  I was just getting my terms mixed up RE course/class/lesson/section

Comment (canyon289) [reply lesson_2_Section_30_new.ipynb:1]: Ah sorry, I should write this down somewhere
Course - > The whole thing, a collection of lessons, as well as exercises, and lesson references
Lesson -> A collection of sections
Section -> One individual video

Comment (bwengals) [reply lesson_2_Section_30_new.ipynb:1]: <p>I think I'm both 1 and 2.  I can see different folks coming at it from either direction.  If youre a SQL whiz, this is like window functions, or maybe you're an engineer and you're thinking about signal processing or spectral stuff.  If you've taken machine learning 101 somewhere, you are maybe thinking "is this not KNN?"  If neither, 'nearest neighbor' at least has a more literal interpretation, where 'window function' might sound a bit mysterious.  I think at least in a video however it's explained it should be mentioned at least as a parenthetical "by the way yes, this is KNN regression".</p>

Comment (canyon289) [reply lesson_2_Section_30_new.ipynb:1]: great refinement, Ill get that added into the script in the right place

Comment (canyon289) [reply lesson_2_Section_30_new.ipynb:1]: > Could make an exercise to implement k nearest neighbor regression from scratch in python since its easy, or use scikit-learn on some data, or both and compare? 

Good idea, I do it a very hacky way. We can get the students do it using vectorization

Comment (bwengals) [reply lesson_2_Section_30_new.ipynb:1]: gotcha, sounds good

**Comment by bwengals** [comment]
Ready for merge?  If so pls do I think it looks great
